### PR TITLE
Better support for symbolic ranges

### LINKED ISSUE #18400: imageset over Range does not work
Simple example: `imageset(lambda x: x*2, Range(n))`
It throws the exception `ValueError: invalid method for symbolic range` while it tries to check `self.size == 1` where `self` is the `Range` object.

### PR DESCRIPTION
**Pull request by oscargus**
#### References to other Issues or PRs

Closes #18400

#### Brief description of what is fixed or changed
I started playing around with symbolic ranges and this seems to pass all the tests I ran at home.

If this seems like a way forward, I will add tests for #18400. (And try to fix any failing test...)

#### Release Notes

* sets
   * Better support for symbolic `Range`.

### CODE REVIEW

**Comment by oscargus** [comment]
I understand this is by convention, but not really obvious anyway (and a bit annoying here):
```
In [29]: oo.is_positive
Out[29]: False

In [30]: oo > 0
Out[30]: True
```

**Comment by oscarbenjamin** [comment]
```
In [1]: oo.is_extended_positive
Out[1]: True
```
New docs explaining this:
https://docs.sympy.org/dev/guides/assumptions.html#predicates

**Comment by oscargus** [comment]
Thanks! Forgot about those extended ones...

**Comment by oscargus** [comment]
One may want to consider a `SymbolicRangeError` which inherits from `ValueError` (or maybe `TypeError` sometimes that is thrown as well).

File: sympy/sets/tests/test_fancysets.py [hunk 1]
```diff
@@ -848,6 +848,55 @@ def test_ComplexRegion_contains():
...
+    assert Range(n, n+1)[0] == n
+    assert Range(n).size == n
+    assert Range(n+1).size == n+1
+    assert Range(n, n+1).size == 1
+
+    n = Symbol('n', integer=True, positive=True)
+    assert Range(n)[0] == 0
+    assert Range(n, n+1)[0] == n
+    assert Range(n).size == n
+    assert Range(n, n+1).size == 1
```
Comment (smichr) [inline sympy/sets/tests/test_fancysets.py:878]: nice progression of tests!

File: sympy/sets/fancysets.py [hunk 2]
```diff
@@ -895,7 +898,7 @@ def __getitem__(self, i):
             if not (all(i.is_integer or i.is_infinite
                     for i in self.args) and ((self.stop - self.start)/
                     self.step).is_extended_positive):
-                raise ValueError('invalid method for symbolic range')
+                raise ValueError('Tnvalid method for symbolic Range')
```
Comment (smichr) [inline sympy/sets/fancysets.py:901]:
```suggestion
                raise ValueError('Invalid method for symbolic Range')
```

**Comment by smichr** [comment]
lgtm

**Pull request merged**